src/DynamicNLPModels.jl

```diff
@@ -205,39 +235,49 @@ function LQDynamicModel(dnlp::LQDynamicData{T,S,M}; condense = false) where {T,S
     Qf = dnlp.Qf
     ns = dnlp.ns
     nu = dnlp.nu
+    E  = dnlp.E
+    F  = dnlp.F
 
     sl = dnlp.sl
     su = dnlp.su
     ul = dnlp.ul
     uu = dnlp.uu
+    gl = dnlp.gl
+    gu = dnlp.gu
 
 
     if condense == false
-        H = _build_H(Q, R, N; Qf=Qf)
-        J = _build_J(A, B, N)
+        H  = _build_H(Q, R, N; Qf=Qf)
+        J  = _build_J(A, B, N)
+        EF = _build_EF(E, F, N)
 
         c0 = 0.0
 
         nvar = ns * (N + 1) + nu * N
-        nnzj = length(J.rowval)
-        nnzh = length(H.rowval)
-        ncon = size(J, 1)
 
         c  = zeros(nvar)
 
-        lvar = copy(s0)
-        uvar = copy(s0)
-        ucon  = zeros(ncon)
-        lcon  = zeros(ncon)
+        lvar  = copy(s0)
```

Comment:
try to minimize the use of `vcat`. A better way is to first create an array of proper size, and set the entry of the array